lib/internal/bootstrap/node.js

```diff
@@ -246,6 +247,60 @@ if (!config.noBrowserGlobals) {
     return performance;
   });
 
+  defineOperation(globalThis, 'prompt', function prompt(
+    message = 'prompt',
+    defaultValue
+  ) {
+    const { validateString } = require('internal/validators');
+    validateString(message);
+    if (defaultValue !== undefined) {
+      validateString(defaultValue);
+    }
+
+    defaultValue ??= null;
+    if (!process.stdin.isTTY) {
+      return null;
+    }
+
+    process.stdout.write(message + '\r\n');
+    const res = readFromStdInSync();
+    return res || defaultValue;
+
+    // Mimic implementation of `prompt` of deno
+    function readFromStdInSync() {
+      const fs = require('fs');
+      const LF = '\n'.charCodeAt(0);
+      const CR = '\r'.charCodeAt(0);
+      const charaBuf = new Uint8Array(1);
+      const stdfd = process.platform === 'win32' ?
+        process.stdin.fd :
+        fs.openSync('/dev/tty', 'r');
+      const res = [];
+
+      while (true) {
+        const bytesRead = fs.readSync(stdfd, charaBuf, 0, 1);
+        if (bytesRead === 0) {
+          break;
+        }
+        if (charaBuf[0] === CR) {
+          const bytesRead = fs.readSync(stdfd, charaBuf, 0, 1);
```

Comment:
This kind of implementation is going to be inherently flaky in cases in which `process.stdin` is also used for reading.